File: CS50x 2023 Solutions/Week 6/ProbSet 6/readability.py
from cs50 import get_string
from sys import exit

words = 1
letters = 0
sentences = 0

text = get_string("Text: ")

for c in text:
    if c.isalpha() == True:
        letters += 1
    elif c == " ":
        words += 1
    elif c in ["!", "?", "."]:
        sentences += 1

index = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8

if index < 1:
    print("Before Grade 1")
    exit(0)

index = round(index)

if index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: CS50x 2023 Solutions/Week 2/ProbSet 2/readability.c
#include "cs50.h"
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int get_clindex(string text);
float clindex;

int main(void)
{

    string text = get_string("Text: ");

    get_clindex(text);

    if (clindex < 1)
    {
        printf("Before Grade 1\n");
    }

    clindex = round(clindex);

    if (clindex > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        for (int j = 0; j <= clindex; j++)
        {
            if (clindex == j)
            {
                printf("Grade %i\n", j);
            }
        }
    }
}

int get_clindex(string text)
{
    float letters = 0, words = 1, sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    clindex = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8;

    return clindex;
}